packages/react/__tests__/rehydration.js

```diff
@@ -166,3 +197,131 @@ test('initializing another Emotion instance should not move already moved styles
     </head>
   `)
 })
+
+test('xxx', () => {
+  const { app, styles } = disableBrowserEnvTemporarily(() => {
+    resetAllModules()
+
+    let cache = createCache({ key: 'mui' })
+    let { extractCritical2, constructStyleTags } = createEmotionServer(cache)
+
+    const rendered = ReactDOMServer.renderToString(
+      <CacheProvider value={cache}>
+        <Global styles={{ body: { color: 'white' } }} />
+        <Global styles={{ html: { background: 'red' } }} />
+        <main css={{ color: 'green' }}>
+          <div css={{ color: 'hotpink' }} />
+        </main>
+      </CacheProvider>
+    )
+    const extracted = extractCritical2(rendered)
+    return {
+      app: extracted.html,
+      styles: constructStyleTags(extracted)
+    }
+  })
+
+  safeQuerySelector('head').innerHTML = styles
+  safeQuerySelector('body').innerHTML = `<div id="root">${app}</div>`
+  expect(safeQuerySelector('html')).toMatchInlineSnapshot(`
+    <html>
+      <head>
+        <style
+          data-emotion="mui-global l6h"
+        >
+          body{color:white;}
+        </style>
+        <style
+          data-emotion="mui-global 10q49a4"
```

Comment:
I've also noticed that global styles (with `-global` suffix) are not put on the inserted cache while rehydrating because this only checks the literal key:
https://github.com/emotion-js/emotion/blob/4d7efcb33c5e70f6f1732a96023a25f5270e72dd/packages/cache/src/index.js#L90-L96

I don't know what consequences this have - we'll have to check this.

Reply:
cc @mitchellhamilton , the inserted/registered stuff is fuzzy to me, so your input would be very much appreciated on this one

Reply:
This is correct. When on the client, global styles(as in Global component, not injectGlobal) are not put in the inserted cache. The global styles are rehydrated by the Global component rather than the cache https://github.com/emotion-js/emotion/blob/6c4a9e50f177900f54f7b3368a55b1a7d5e3c002/packages/react/src/global.js#L92-L110

Reply:
Hm... it was actually asked for that `injectGlobal` styled could be removable. This is, of course, not a concern of this PR - but I was also thinking that we could support this in the future.